imgix/urlbuilder.py

```diff
@@ -52,13 +52,30 @@ def __init__(
         if not isinstance(domains, (list, tuple)):
             domains = [domains]
 
+        self.validate_domain(domains)
         self._domains = domains
         self._sign_key = sign_key
         self._use_https = use_https
         self._shard_strategy = shard_strategy
         self._shard_next_index = 0
         self._sign_with_library_version = sign_with_library_version
 
+    def validate_domain(self, domain):
+        regex = re.compile(
```

Comment:
Are we taking a performance hit by compiling this regex here rather than compiling it once globally and re-using it?

Reply:
Are we re-using it if the object is only initialized once?

Reply:
The client is only initialized once, but it can be given multiple domains and multiple clients could be initialized.

Reply:
Makes sense. Also, in case you were actually wondering about the [performance part](https://docs.python.org/3/library/re.html#re.compile) of your question.

> but using `re.compile()` and saving the resulting regular expression object for reuse is more efficient when the expression will be used several times in a single program.